### ISSUE TITLE
Checks not reported when using--stdin flag and piping a file

### ISSUE THREAD

**Issue by canatella** [pinterest/ktlint#1683]
## Expected Behavior

When piping a kotlin file with some errors and using the `--stdin` argument, the errors should be reported by ktlint.

## Observed Behavior

No error is displayed. When the same code is put into a file and it is passed as an argument without using `--stdin`, it works correctly.

## Steps to Reproduce

```
# Piping directly with --stdin, nothing is reported:
$ printf "package foobar\n\n\n" | ktlint --stdin

# Storing in a file and checking the file works:
$ printf "package foobar\n\n\n" > ktlint-test.kt
$ ktlint ktlint-test.kt 
ktlint-test.kt:3:1: Needless blank line(s) (no-consecutive-blank-lines)
```
## Your Environment

* Version of ktlint used: 0.45.2
* Relevant parts of the `.editorconfig` settings: defaults
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): n/a
* Version of Gradle used (if applicable): n/a
* Operating System and version: Ubuntu 22.04.1 LTS

**Comment by paul-dingemans** [comment]
I can't get your example working. But next one does give output:
```
$ printf "package foobar\n\n\n\nimport foo.*\n" | ktlint-0.45.2 --stdin
<stdin>:3:1: Needless blank line(s) (no-consecutive-blank-lines)
<stdin>:5:1: Wildcard import (cannot be auto-corrected) (no-wildcard-imports)
```
I have no reasonable explanation why this happends.

**Comment by canatella** [comment]
The `printf "package foobar\n\n\n" | ktlint --stdin` is just a way I found to reproduce it. It also happens here on other, more common files.

**Comment by paul-dingemans** [comment]
Please provide more details on other examples in which it does not work for other rules than `no-consecutive-blank-lines`.

**Comment by canatella** [comment]
Indeed I can't. If I remember correctly, the issue that was not detected in a "normal" file might indeed also have been a consecutive blank line.

**Comment by canatella** [comment]
I mean there is obviously a bug right ? Here is another example:
```
package foo.bar



class MyClass() {
    val something: Int
}
```

It seems to only happen for consecutive lines at the top scope, this works correctly:

```
package foo.bar

class MyClass() {


    val something: Int
}
```
